### ISSUE TITLE
UnknownContentCollectionError when Soft Links Used in Content Collections

### ISSUE THREAD

**Issue by equt** [withastro/astro#8262]
### Astro info

```block
Astro version            v2.10.14
Package manager          npm
Platform                 darwin
Architecture             arm64
Adapter                  Couldn't determine.
Integrations             None or couldn't determine.
```

### What browser are you using?

Chrome

### Describe the Bug

1. Create a collection, either in `data` or in `content`
2. **Soft** link files or entire directory into the `src/content` directory
3. Query that collection

### What's the expected result?

It should work as normal

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-x2fbkv?file=src%2Fpages%2Findex.astro&on=stackblitz

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by equt** [comment]
I can't figure out a way to create a real soft link in the above reproducible sandbox. But the problem is cause by the

https://github.com/withastro/astro/blob/04fa5a14536aca01dbf545f972b9883797730535/packages/astro/src/content/utils.ts#L182-L197

This utility function will parse the collection name out of the `entry`, which comes from the `fileId`. Unfortunately, Vite will resolve the (link) path before Astro ever has a chance to read the it.

**Comment by natemoo-re** [comment]
Yep, this makes sense and it seems like we should support it! Probably didn't account for symlinks.

**Comment by YonicDev** [comment]
While tackling this I found out that there's an option for Vite to preserve soft links when resolving.
Add the following config in `astro.config.mjs` and it should work:
```js
export default defineConfig({
  vite: {
    resolve: {
      preserveSymlinks: true
    }
  }
})
```

**Comment by ascorbic** [comment]
This should have been fixed by #11236
